config/config_test.go

```diff
@@ -129,3 +129,29 @@ func TestParseEndpoint(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportIP(t *testing.T) {
+	// Host is ip address, enable virtual host style
+	c := Config{
+		AccessKeyID:            "AccessKeyID",
+		SecretAccessKey:        "SecretAccessKey",
+		Endpoint:               "https://192.168.0.1:443",
+		EnableVirtualHostStyle: true,
+	}
+	err := c.parseEndpoint()
+	assert.Nil(t, err)
+	err = c.Check()
+	assert.NotNil(t, err)
+
+	//
+	c = Config{
+		AccessKeyID:     "AccessKeyID",
+		SecretAccessKey: "SecretAccessKey",
+		Endpoint:        "https://2001:db8::68:443",
```

Comment:
IPv6 should be `[2001:db8::68]` here?

Reply:
yes, eg: `The string s can be in IPv4 dotted decimal ("192.0.2.1"), IPv6 ("2001:db8::68"), or IPv4-mapped IPv6 ("::ffff:192.0.2.1") form. `